[DCOS-44856] Release Notes for Spark 2.5.0-2.2.1-3 (#1759)
mesosphere/dcos-docs-site: pages/services/spark/2.5.0-2.2.1/release-notes/index.md

---
layout: layout.pug
navigationTitle: Release Notes
title: Release Notes
menuWeight: 145
excerpt: Release notes for DC/OS Apache Spark 2.5.0-2.2.1
featureMaturity:
---

## Version Spark and Spark History 2.5.0-2.2.1

### New Features
- Unique Mesos Task IDs for Spark executors
- Trusted Ubuntu 18.04 base Docker image

> - `nobody` user support on RHEL/CentOS (via configuration)

Reviewer: Compare previous release notes to this version. These should be worded similarly eg 
- `Added unique Mesos Task IDs for Spark executors`
- `Added trusted Ubuntu 18.04 base Docker image`... etc

Same for updates:
- `Updated JRE to 1.8.192`
- `Upgraded Ubuntu to 18.04`

Author: Done

### Updates
- Changed default user for Docker container from ‘root’ to ‘nobody’

> - Updated JRE upgrade to 1.8.192

Reviewer: ```suggestion
- Updated JRE to `1.8.192`
```

> - Updated Ubuntu upgrade 18.04

Reviewer: ```suggestion
- Upgraded to Ubuntu 18.04
```

- Hadoop dependencies update from 2.7.3 to 2.7.7 (fixes CVE-2016-6811, CVE-2017-3162, CVE-2017-3166, CVE-2018-8009)
- Jetty dependencies update from jetty 9.3.11.v20160721 to 9.3.24.v20180605 (fixes CVE-2017-7658)
- Jackson dependencies update from 2.6.5 to 2.9.6 (fixes CVE-2017-15095, CVE-2017-17485, CVE-2017-7525, CVE-2018-7489, CVE-2016-3720)
- ZooKeeper dependencies update from 3.4.6 to 3.4.13

### Bug fixes
- `dcos task log` now works because of unique Mesos Tasks IDs of Spark executors
- Fixed unstable health checks for Spark Dispatcher and History Server
- Spark dispatcher task output now redirected to stdout and available in logs

### Breaking Changes

> - new configuration option added: ‘docker_user’ which allows to override a user when running Spark using Docker containerizer

Reviewer: Capitalize and clean up a little. Suggestion:

```
We've added a new configuration option: `docker_user` which enables overriding the default Spark user inside Docker containerized jobs
```

Author: Done

> - We've added a new configuration option added: ‘docker_user’ which allows to override a user when running Spark using Docker containerizer

Reviewer: ```suggestion
- We've added a new configuration option: ‘docker_user’ which allows overriding the user when running Spark using Docker containerizer
```

> - Spark default Hadoop dependency is updated from 2.6 to 2.7

Reviewer: Suggestion: `The default Hadoop dependency is now 2.7, not 2.6`

Author: Done
